src/codeql.ts

```diff
@@ -280,3 +320,31 @@ function getDatabaseSHA(database: string): string {
     return "HEAD";
   }
 }
+
+/**
+ * Gets the query for a pack, assuming there is a single query in that pack's default suite.
+ *
+ * @param codeql The path to the codeql CLI
+ * @param queryPack The path to the query pack on disk.
+ * @returns The path to a query file.
+ */
+async function getRemoteQueryPackDefaultQuery(
+  codeql: string,
+  queryPack: string
+): Promise<string> {
+  const output = (
+    await getExecOutput(codeql, [
+      "resolve",
```

Comment:
You could use `--format=json` which returns a JSON array of strings. Parsing JSON might be more robust than splitting by `\n` . It's unlikely the CLI would ever produce and old-style MacOS line ending (`\r` without `\n`) or a query path would contain a newline, but still.

Reply:
Thanks, that is much better! I didn't think to check that this command supported the format option.